### DISCUSSION TITLE
Determination of Parameters

### DISCUSSION THREAD

**Discussion by adlai** [zai-ac/scalpl#5] [Polls]
The codebase is replete with magic numbers; some of these are critical, while others may be considered the space of reasonable discourse without straying from the strategically optimal path.

**Comment by roks0n** [comment]
What are these magic numbers you speak of? Settings for trading parameters? Constants? Do they need tweaking often?

**Reply by adlai** [reply]
some are constants, like API keys; some are trading parameters, like the rate limiting preference of brokers, bucket shops, and limit order exchanges; and some are not resolved yet.

**Reply by roks0n** [reply]
Moving discussion here:

@adlai  replied:
> There have been several profit opportunities where derivatives could be arbitraged; however, this is not possible using the arithmetic schemas autogenerated, thus, I propose moving our discussion of higher mathematics to either github, or at least, postponing until after the weekend.

Me continuing ...

So what are these parameters? If they are specific to a particular ticker maybe they can be provided as cli args?

**Reply by adlai** [reply]
> are these parameters?

the most obvious, number of separate orders, are definitely parameters for the complex algorithm that get collapsed down to minimal size by bucket shops; trading engines also do this, at high frequency.

**Reply by roks0n** [reply]
Do you have multiple copies of scalpl on a machine? Each for different exchange / ticker?

Eg:
- 1 copy for: binance BTCWBTC
- 1 copy for: binance USDTUSDC
- 1 copy for: bybit BTCWBTC

Above examples are just for illustrative purposes.

**Reply by adlai** [reply]
> have multiple copies of scalpl on a machine?

only after getting berated for not arbitraging all possible markets simultaneously...")

**Reply by roks0n** [reply]
I see several solutions:

1. Have a copy of scapl per exchange and accept symbol-specific settings as cli args?
2. Have one copy of scapl which then either:
  2.1. accepts symbol-specific and exchange-specific settings as cli args
  2.2. accepts symbol-specific settings as cli args AND a path to a exchange-specific config file


So 2.2 would be: `scapl --exchange-config=<path to file> --ticker=BTCWBTC --<other params>`

**Reply by adlai** [reply]
> Above examples are just for illustrative purposes.

the most painful are the triangular arbitrages, found across all possible markets!

You could earn the most by doing triangular arbitrages without much loss on derivatives if you only place limit orders in derivatives as cost of signal reading.

**Reply by roks0n** [reply]
> triangular arbitrages

I was reading about these just yesterday! :D Do we know how many of such opportunities are out there? Which symbols would we use for this?

**Reply by adlai** [reply]
If zai-ac members keep on moping that most volume is spoofed, then there will never be any worthwhile opportunities. Real liquidity is there.

**Reply by roks0n** [reply]
Surely this can be tested?
